app/javascript/admin/controllers/config_controller.js

```diff
@@ -1,9 +1,64 @@
 import { Controller } from 'stimulus';
 
 const recaptchaFields = document.querySelector('#recaptchaContainer');
+const emailSigninAndLoginCheckbox = document.querySelector(
+  '#email-signup-and-login-checkbox',
+);
+const emailAuthSettingsSection = document.querySelector(
+  '#email-auth-settings-section',
+);
+const modalAnchor = document.querySelector('.admin-config-modal-anchor');
+const emailAuthModalTitle = 'Disable email address registration';
+const emailAuthModalBody =
+  '<p>If you disable email address as a registration option, people cannot create an account with their email address.</p><br /><p>However, people who have already created an account using their email address can continue to login.</p>';
+const emailAuthModalBodyInviteOnly =
+  '<p>If you disable email address as a registration option, people cannot create an account with their email address. This will also disable <strong>invite-only mode</strong>.</p><br /><p>However, people who have already created an account using their email address can continue to login.</p>';
+
+const adminConfigModal = (
```

Comment:
So I am not sure of my implementation of the modal will pass QA 🙈  but let me explain my thinking:

- I thought to make the modal implementation as general as possible, so that if other refactors of the Admin/Config page require a modal, we can use this one. I basically make all variable modal content passable as parameters.
- Regarding the text content passed into the modal, I thought to use the `Constants::SiteConfig::DETAILS` file that we have on the Rails side (and define the text content in that file), but couldn't figure out how to read the file.

Reply:
Ah, another modal dialog box! I wonder if there's a way we could use the non-admin modal for this. In any case: looks good!

Reply:
I didn't investigate this but I'm curious if the regular crayons modal (`app/javascript/crayons/Modal/Modal.jsx`) was usable at all here? It would be ideal to reuse it, like @joshpuetz mentioned.

If we decide that it is not workable to reuse that modal, then I would suggest that ultimately we move the `adminConfigModal` out of this file (and then import it here) so that we can re-use it across the admin sections of the app. I haven't used stimulus js before, but I'd hope that this would Just Work™️. Right now, the `adminConfigModal` is coupled to this specific `config` controller, and if we decide that we can't use the crayons modal, it would be ideal to make this modal as reusable as possible for the admin section of the app.

But I think that could be a separate PR if need be.

Reply:
@vaidehijoshi These are awesome points; I've created a note in Team Email Login to address this as a separate PR 👍